### ISSUE TITLE
Hyper opens off screen, cannot move it back to the screen (Windows 10)

### ISSUE THREAD

**Issue by markvincze** [zeit/hyper#1682]
- [x] I am on the [latest](https://github.com/zeit/hyper/releases/latest) Hyper.app version
- [x] I have searched the [issues](https://github.com/zeit/hyper/issues) of this repo and believe that this is not a duplicate

- **OS version and name**: Windows 10
- **Hyper.app version**: 1.3.1
- **Link of a [Gist](https://gist.github.com/) with the contents of your .hyper.js**: https://gist.github.com/markvincze/5d490b94871810990b5560ead95548ff
- **The issue is reproducible in vanilla Hyper.app**: Yes, same thing happens if I remove all the plugins.

## Issue
I used hyper on my laptop when external screens were connected. Now I'm trying to use it again, and now I don't have any external screens connected, and Hyper seems to open off-screen. I see its thumbnail if I do alt-tab, but then it doesn't show up on the screen. And if I minimize it (click on it again on the taskbar), I can see that it floats down from off-screen. (Maybe this is not a Hyper-specific problem, but I haven't encountered this with any other apps, although I use some other electron-based apps too.)

Things I tried:
 - Press Win+Shift+Arrow and Win+Arrow to move the window to the right screen, but nothing happens.
 - Shift+Right click on the task bar to be able to click on "Move" in the context menu, but the context menu does not open.
 - Uninstalled and reinstalled Hyper.

Any idea or workaround would be appreciated!

**Comment by jamo** [comment]
There is this trick that allows you to move windows just with keyboard. It's been like 5 years since I did and I dont have a windows to try it with.

But it was along the lines of: `alt + space` opens the right click menu for the currently active window. In there you could navigate with arrow keys and select `Move`.
And then you could use arrow keys to move it over to correct screen.

So you could try it for a window that is visible and then do it for hyper.
Otherwise iirc the window locations are somewhere in registry... :D

**Comment by markvincze** [comment]
@jamo, Awesome, this worked! :)

Although interestingly the Move command did not help, it showed the Move cursor with the arrows, and I tried to click and drag, but nothing happened.
On the other hand, the Maximize command moved back the window to the screen, so I'm saved.

Thanks!

**Comment by paultyng** [comment]
FWIW I had this same issue, not sure if its something that can be reopened for addressing? The workaround works, but my laptop frequently changes between multimonitor when docked to single monitor and this problem reoccurs frequently.

**Comment by justrhysism** [comment]
@paultyng I think this is a duplicate of https://github.com/zeit/hyper/issues/1608

**Comment by nitech** [comment]
If you've installed something that captures ALT+Space, you can use ALT+SHIFT+Space instead.

**Comment by deanhtid** [comment]
This has been driving me crazy for months. Finally found the solution. Thanks, guys!

**Comment by nitech** [comment]
@nitech, this is nitech from the future. Thanks. Really helped me.

**Comment by ichikai-taichou** [comment]
I know this is a bit late, but the above didn't work for me.

What I did was precisely the same as above (the ALT + Space) - but the VM window sceen didn't take the command.

Shift + Right Clicked on the connected VM icon on the task bar, and selected Maximise.
Worked.

**Comment by Yloganathan** [comment]
Selecting the app and then win + left arrow brought the hyper terminal back to the main screen for me.

**Comment by albinekb** [comment]
We should try to find the root cause of this error, anyone know how to reproduce? How did you get hyper off screen?

**Comment by albinekb** [comment]
Ah #1608

**Comment by albinekb** [comment]
Being fixed in https://github.com/zeit/hyper/pull/2459

**Comment by ArslanAmeer** [comment]
Solution from @kevlut  worked for me.

> This may be helpful if you see it open in the toolbar like I was, it worked for me: zeit/hyper#1682
TL;DR with the github for desktop window "focused" hit **`Alt` +` spacebar`** to open the window context menu on screen and maximize the window. Not sure if this is related, but may be something with electron.

i can see github desktop window when i hover on taskbar icon.
Hover then press `Alt` + `spacebar` and click maximize. Simple! i dont know exactly how and why this is happening but it worked for me 👍 !

**Comment by RiikRakJones** [comment]
@jamo I registered here, simply to thumbs up this man, I've been trying to fix this issue for about 4 hours. This solved it in 10 seconds. Thank you

**Comment by chabou** [comment]
@RiikRakJones It has been definitely fixed in our current canary (upcoming v2). Sorry for your 4 hours 😞

**Comment by hasparus** [comment]
Alt+Space > Move worked for me :D Thanks 👍

**Comment by BrysonR** [comment]
Alt + Space > Maximize worked great for me.

**Comment by kevlambrecht** [comment]
Alt + Space > Maximize.  And then grab tool bar and slide window to get out of full screen on your main display.

**Comment by nitech** [comment]
This is Nitech from even further in the future. Looks like we have a final solution. When you right click in Windows 10 you now have the option of: "Move Slack to Current Display"

![image](https://user-images.githubusercontent.com/1279436/47716618-91182b80-dc43-11e8-858d-9d8c2374a635.png)

**Comment by vodafone001** [comment]
> - [x] I am on the [latest](https://github.com/zeit/hyper/releases/latest) Hyper.app version
> - [x] I have searched the [issues](https://github.com/zeit/hyper/issues) of this repo and believe that this is not a duplicate
> 
> - **OS version and name**: Windows 10
> - **Hyper.app version**: 1.3.1
> - **Link of a [Gist](https://gist.github.com/) with the contents of your .hyper.js**: https://gist.github.com/markvincze/5d490b94871810990b5560ead95548ff
> - **The issue is reproducible in vanilla Hyper.app**: Yes, same thing happens if I remove all the plugins.
> 
> ## Issue
> I used hyper on my laptop when external screens were connected. Now I'm trying to use it again, and now I don't have any external screens connected, and Hyper seems to open off-screen. I see its thumbnail if I do alt-tab, but then it doesn't show up on the screen. And if I minimize it (click on it again on the taskbar), I can see that it floats down from off-screen. (Maybe this is not a Hyper-specific problem, but I haven't encountered this with any other apps, although I use some other electron-based apps too.)
> 
> Things I tried:
>  - Press Win+Shift+Arrow and Win+Arrow to move the window to the right screen, but nothing happens.
>  - Shift+Right click on the task bar to be able to click on "Move" in the context menu, but the context menu does not open.
>  - Uninstalled and reinstalled Hyper.
> 
> Any idea or workaround would be appreciated!

**Comment by Tirotu** [comment]
Same issue here, I have a laptop and every time I come to work if I left the VM open I have to close it and reopen it to move it to the correct screen, or else whenever it is not Maximized it is off the screen.  I did eventually find a better workaround so I don't have to close it down.  Do the right click and move, then hold Shift + CTRL + ALT and move with the arrow keys.  Hope this helps.
